network/p2p/peerManager.go

```diff
@@ -105,19 +103,18 @@ func (pm *PeerManager) updatePeers() {
 		Msg("connecting to peers")
 
 	// ask the connector to connect to all peers in the list
-	failedIDs := pm.connector.UpdatePeers(pm.ctx, ids)
-
-	err = failedIDMapToSingleError(failedIDs)
-	if err != nil {
-		pm.logger.Error().Int("failed_peer_count", len(failedIDs)).Err(err).Msg("failed to connect to peers")
+	err = pm.connector.UpdatePeers(pm.unit.Ctx(), ids)
+	if err == nil {
+		return
 	}
-}
 
-// failedIDMapToSingleError converts the failedIDs map to a single error
-func failedIDMapToSingleError(failedIDs map[flow.Identifier]error) error {
-	multierr := new(multierror.Error)
-	for id, err := range failedIDs {
-		multierr = multierror.Append(multierr, fmt.Errorf("failed to connect to %s: %w", id.String(), err))
+	var e *UnconvertableIdentitiesError
+	if errors.As(err, &e) {
```

Comment:
The pattern we've been using for this is defining an error type-checking function along-side the error type. Here's an example from protocol state package: https://github.com/onflow/flow-go/blob/9d322fbe576b1095c177b9a7ffb61cbc39a78c4a/state/errors.go#L7-L35